### PR TITLE
fix: typed error for reissue external notes

### PR DESCRIPTION
**Pull request by maan2003**

### CODE REVIEW

**Review by dpc — APPROVED** [review]

File: modules/fedimint-mint-client/src/lib.rs [hunk 1]
```diff
@@ -743,6 +743,14 @@ impl ClientModule for MintClientModule {
     }
 }
 
+#[derive(thiserror::Error, Debug, Clone)]
+pub enum ReissueExternalNotesError {
+    #[error("Federation ID does not match")]
+    DifferentFederationId,
```
Comment (dpc) [inline modules/fedimint-mint-client/src/lib.rs:749]: "wrong" would be a better word than "different", imo

File: modules/fedimint-mint-client/src/lib.rs [hunk 2]
```diff
@@ -743,6 +743,14 @@ impl ClientModule for MintClientModule {
     }
 }
 
+#[derive(thiserror::Error, Debug, Clone)]
+pub enum ReissueExternalNotesError {
```
Comment (tvolk131) [inline modules/fedimint-mint-client/src/lib.rs:747]: What if we added a third error type?

```
    #[error("Untyped error")]
    Other(anyhow::Error)
```

Then we could change the return type of `reissue_external_notes()` to `Result<OperationId, ReissueExternalNotesError>` and remove this line:

```/// Can return error of type [`ReissueExternalNotesError`]```

Comment (maan2003) [reply modules/fedimint-mint-client/src/lib.rs:747]: I think .downcast is fine, Other cause error chain issues

**Review by dpc — APPROVED** [review]

**Comment by fedimint-backports** [comment]
Successfully created backport PR for `releases/v0.3`:
- #5245

**Pull request merged**